components/dashboard/src/data/setup.tsx

```diff
@@ -127,3 +129,60 @@ function createIDBPersister(idbValidKey: IDBValidKey = "gitpodQueryClient"): Per
         },
     };
 }
+
+const supportedMessages = new Map<string, typeof Message>();
+
+function initializeMessages() {
+    const constr = [
+        ...Object.values(require("@gitpod/public-api/lib/gitpod/experimental/v2/organization_pb")),
+        ...Object.values(require("@gitpod/public-api/lib/gitpod/experimental/v2/workspace_pb")),
+        ...Object.values(require("@gitpod/public-api/lib/gitpod/experimental/v2/pagination_pb")),
+    ];
+    for (const c of constr) {
+        if ((c as any).prototype instanceof Message) {
+            supportedMessages.set((c as any).typeName, c as typeof Message);
+        }
+    }
+}
+initializeMessages();
+
+export function dehydrate(message: any): any {
```

Comment:
Do you mean we rather have a wrapper around `useQuery` something like `useMessageQuery<T extends Message>` which will make sure that toJsonString() and fromJsonString is called?

Reply:
I'm also not happy with that special marshalling logic, but I couldn't find an alternative that I liked better. The underlying issue is that our new API doesn't expose simple JSON objects but javascript objects with additional methods on them. 
I considered the following alternatives:
- write our own proto generator that creates JSON objects and not GRPC messages. This would make working with our API generally more convenient and idiomatic, but comes with big additional effort in writing and maintaining the code generator.
- manually translates all types to JSON in the API layer. basically the same but manually. not an option, becuase it is more effort and more error prone.
- use messages and teach the react-query to (de)hydrate them (the solution in this PR)

Reply:
I’m not very familiar with the interface of the grpc client or messages, but basically ensuring we only work with json objects and not class instances as query results. This removes serialization problems and in general I believe makes our state simpler to work with and less coupled.

Is there an option with the client to have it produce json objects and not class instances?

Reply:
We discussed that we are going with this solution but spend timeboxed effort on a code generator that would allow us to use idomatic TS shapes that are serializable through JSON.stringify/parse